### DISCUSSION TITLE
Add second variable (Nedelec first kind) to boundary edge dofs only

### DISCUSSION THREAD

**Discussion by qianhanhua** [libMesh/libmesh#3292] [Q&A] [answered]
Hi, all, I wanted to try a different formulation for my Mie Scattering model to see whether it produces better results.  It requires to add a second variable, i.e. the tangential H field for boundary edge dofs when imposing boundary conditions. 

I studied about the relevant examples: `systems_of_equations_ex1` and `subdomains_ex2`, but didn't find ways to achieve what I needed. I did not want to add variable for the entire boundary element/Tetrahedron, as it will create variables for the inner edges as well. Does libmesh support adding variable to only part of the dofs in an finite element?  Thank you.

**Comment by roystgnr** [comment, marked as answer]
There's no direct support for adding a variable to only part of an element, but you can "fake it": libMesh lets you combine multiple elements of different dimension - e.g. a mesh can include both a Tet10 and a Tri6 sharing the 6 nodes on one of the tet's sides - and if you give them different subdomain ids, that allows you to define variables that are only active on one but not the other.  There are a few unit tests and IIRC an example you might look at, but I'm afraid nothing very advanced.

**Reply by qianhanhua** [reply]
Thank you @roystgnr, I think that should be the direction I will work towards. By the way, can you give me some reliable example links to such implementation? I didn't find relevant examples [here](https://libmesh.github.io/examples.html). A quick google search didn't give me much useful results either.

**Reply by roystgnr** [reply]
Let me see ... the relevant unit tests are in tests/mesh/mixed_dim_mesh_test.C ... but I can't find the example I thought I was remembering either.  Maybe I skimmed over them too quickly, or maybe I'm remembering something from an application rather than from the library itself.  Either way, sorry.

**Reply by qianhanhua** [reply]
@roystgnr No problem. The info that you have given is already very helpful. Thank you!

**Reply by roystgnr** [reply]
> I skimmed over them too quickly

Found it, it's `fem_system_ex4`.  Definitely still nothing advanced, a heat equation on mixed-dimension domain with different thermal conductivities for the interior vs for the shell on one side, but probably at least a little more helpful than the unit tests.

**Reply by qianhanhua** [reply]
Oh, I should go through the examples more carefully. Thank you @roystgnr, will look into it.
